### PR TITLE
558 new nav

### LINKED ISSUE #558: Nav changes per Hugh
Mockup coming.

### PR DESCRIPTION
**Pull request by lschatzkin**
Redoing nav, based on Hugh's input.

Added a modal-type behavior so user can click anywhere outside menu to close menu.

closes #558

### CODE REVIEW

**Comment by esoterik** [comment]
It'd be nice to keep the same mobile behavior as before, I think (the slide-out full length menu rather than the short two links popping up). 

I also wonder if the menu should be hidden (on desktop) when not hovering over it  or the hamburger. To me it feels unexpected when it stays open and a little clunky / intuitive to have to click the hamburger again to close it, but I know there are accessibility concerns with hover-only content. Maybe clicking outside the div with the links should also close the menu?

**Comment by lschatzkin** [comment]
> It'd be nice to keep the same mobile behavior as before, I think (the slide-out full length menu rather than the short two links popping up).
> 
> I also wonder if the menu should be hidden (on desktop) when not hovering over it or the hamburger. To me it feels unexpected when it stays open and a little clunky / intuitive to have to click the hamburger again to close it, but I know there are accessibility concerns with hover-only content. Maybe clicking outside the div with the links should also close the menu?

I discussed the mobile with Hugh, and he likes the menu being the same on mobile and desktop.

I'm not sure about clicking outside the drop down to close. It seems I've mostly seen that with modals. Do you think users would do that? Maybe let's discuss with design folks.

File: app/assets/javascripts/application/nav-close.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+$(document).click(function(event) {
+  $navmenu = $('#nav-menu');
+  $target = $(event.target);
+  if(!$target.closest('#nav-menu').length &&
+  $('#nav-menu').is(":visible")) {
+    $('#nav-menu').css('display', 'none');
+  }
+  else {
+      $('#nav-menu').css('display', 'inline-block');
+  }
```
Comment (vbrown608) [inline app/assets/javascripts/application/nav-close.js]: This file would be easier to read with more conventional whitespace, eg:
```
if (!$target.closest('#nav-menu').length && $('#nav-menu').is(":visible")) {
  $('#nav-menu').css('display', 'none');
} else {
  $('#nav-menu').css('display', 'inline-block');
}
 ```
For javascript, we should use single quotes consistently for strings. In the medium term, we should add a javascript linter to the action center so we can catch these kinds of things automatically.

File: app/assets/javascripts/application/nav-close.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+$(document).click(function(event) {
+  $navmenu = $('#nav-menu');
+  $target = $(event.target);
```
Comment (vbrown608) [inline app/assets/javascripts/application/nav-close.js]: These variables are getting global scope - they should be prefixed with `var`.

**Comment by wioux** [comment]
The menu pops up in an unexpected spot:

Screenshot_2019-07-18_15-18-41

**Comment by wioux** [comment]
The menu is broken if you click anywhere else on the page before opening it.

File: app/assets/javascripts/application/nav-close.js (hunk 1 above)
Comment (lschatzkin) [reply app/assets/javascripts/application/nav-close.js]: Oh, this is some old code. I'm not actually using this, I'm only using css. I'll delete it.

File: app/assets/javascripts/application/nav-close.js (hunk 2 above)
Comment (lschatzkin) [reply app/assets/javascripts/application/nav-close.js]: See above.

**Comment by lschatzkin** [comment]
> The menu is broken if you click anywhere else on the page before opening it.

:(

**Comment by lschatzkin** [comment]
Okay, thanks @wioux for finding that problem. It turns out the legacy jquery code was obstructing the css from working. @vbrown608 ready for re-review.

**Comment by lschatzkin** [comment]
> The menu pops up in an unexpected spot:
> 
> Screenshot_2019-07-18_15-18-41

Oops! Forgot to fix this, doing it now.

**Comment by lschatzkin** [comment]
Ready for review again.

**Comment by vbrown608** [comment]
:dancer: works great!

**Pull request merged**